### ISSUE TITLE
fluvio CLI update is too frequent

### ISSUE THREAD

**Issue by sehz** [infinyon/fluvio#2965]
Note this only affects the development version.  It doesn't apply to the release version.

Every time we invoke CLI,  The version update is performed.  It is good, but this is too frequent. It slows down CLI performance. 

```
fluvio partition list
🔍 Checking for the new version
✅ fluvio-cli is up to date
```

 To optimize this, we should perform the update only in the following scenarios:

`fluvio update`
`fluvio version`

**Comment by EstebanBorai** [comment]
Addressed on #2966

**Comment by sehz** [comment]
@simlay test?

**Comment by tjtelan** [comment]
Confirmed fixed in 0.10.5-alpha.1. We can confirm in the debug output that we only check in update or version

update
```
% RUST_LOG=fluvio_cli::root=debug fluvio update
2023-02-28T20:26:58.599203Z  INFO fluvio_cli::root: Triggered a Fluvio Update Check
🔍 Checking for new version
...
```

version
```
% RUST_LOG=fluvio_cli::root=debug fluvio version
2023-02-28T20:26:53.106189Z  INFO fluvio_cli::root: Triggered a Fluvio Update Check
🔍 Checking for new version
...
```

profile list
```
% RUST_LOG=fluvio_cli::root=debug fluvio profile list
     PROFILE                  CLUSTER                  ADDRESS                         TLS
...
```
